patroni/dcs/etcd.py

```diff
@@ -297,14 +297,20 @@ def _load_machines_cache(self):
 
 
 def catch_etcd_errors(func):
-    def wrapper(*args, **kwargs):
+    def wrapper(self, *args, **kwargs):
         try:
-            return func(*args, **kwargs) is not None
+            return func(self, *args, **kwargs) is not None
         except (RetryFailedError, etcd.EtcdException):
             return False
```

Comment:
Shouldn't be here the same logic (setting `_has_failed` to `True`) as for all other cases?